src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -478,10 +480,79 @@ def createReplicationRule(self, names, rseExpression, scope='cms', copies=1, **k
         except AccessDenied as ex:
             msg = "AccessDenied creating DID replication rule. Error: %s" % str(ex)
             raise WMRucioException(msg)
+        except DuplicateRule as ex:
+            # NOTE:
+            #    The unique constraint is per did and it is checked against the tuple:
+            #    rucioAccount + did = (name, scope) + rseExpression
+            # NOTE:
+            #    This exception will be thrown by Rucio even if a single Did has
+            #    a duplicate rule. In this case all the rest of the Dids will be
+            #    ignored, which in general should be addressed by Rucio. But since
+            #    it is not, we should break the list of Dids and proceed one by one
+            # NOTE:
+            #    This thing here may be slow, because it will wait for Rucio to
+            #    return the history of rules per every Did, but no shorter path exists
+            msg = "A duplicate rule for: \naccount: %s \ndids: %s \nrseExpression: %s\ncopies: %s\n"
```

Comment:
This should be logged before we try to check the rule history (I also think we should mention that the rule history will be checked for X dids. Please make it INFO level

Reply:
I thought this was exactly when it is logged, but yes, I see it is just the message that is created at the proper place but the logging is happening down bellow. I am fixing this now.

Reply:
Done